### ISSUE TITLE
Crash in BSGJSONSerialization.m line 33

### ISSUE THREAD

**Issue by sethfri** [bugsnag/bugsnag-cocoa#899]
### Describe the bug
Crash of unknown origin coming from Bugsnag. We have seen 1 crash coming from 1 user so far.

### Steps to reproduce
Unknown

### Environment
* Bugsnag version: 6.2.3
* CocoaPods version: 1.9.3
* Carthage version (if any): n/a
* iOS/tvOS/macOS version(s): 14.1
* Simulator or physical device: iPhone XR
* Xcode version: 12.1
* Swift version (if applicable): 5.1

### Stack Trace

<details><summary>Stack trace:</summary>

```
Crashed: com.apple.root.default-qos
SIGSEGV 0x0000000cb12dd100
0  libobjc.A.dylib                0x1c3eb69f4 object_getMethodImplementation + 48
1  CoreFoundation                 0x1afdd717c _NSIsNSString + 60
2  Foundation                     0x1b116e2d8 _writeJSONValue + 64
3  Foundation                     0x1b116e24c -[_NSJSONWriter dataWithRootObject:options:] + 88
4  Foundation                     0x1b101ed80 +[NSJSONSerialization dataWithJSONObject:options:error:] + 120
5  MyApp                           0x104bda904 +[BSGJSONSerialization dataWithJSONObject:options:error:] + 33 (BSGJSONSerialization.m:33)
6  MyApp                           0x104bfd8e8 -[BugsnagSystemState sync] + 230 (BugsnagSystemState.m:230)
7  libdispatch.dylib              0x1af9c6fd0 _dispatch_call_block_and_release + 32
8  libdispatch.dylib              0x1af9c8ac8 _dispatch_client_callout + 20
9  libdispatch.dylib              0x1af9cba10 _dispatch_queue_override_invoke + 692
10 libdispatch.dylib              0x1af9d9104 _dispatch_root_queue_drain + 356
11 libdispatch.dylib              0x1af9d98e8 _dispatch_worker_thread2 + 116
12 libsystem_pthread.dylib        0x1f7cbb8cc _pthread_wqthread + 216
13 libsystem_pthread.dylib        0x1f7cc277c start_wqthread + 8
```
</details>

**Comment by johnkiely1** [comment]
Hi @sethfri ,

Thanks for raising, we will investigate and get back to you.

**Comment by johnkiely1** [comment]
@sethfri 
This underlying cause of this issue has been fixed in [v6.2.5](https://github.com/bugsnag/bugsnag-cocoa/releases/tag/v6.2.5).
